fiftyone/plugins/core.py

```diff
@@ -621,24 +668,25 @@ def _iter_plugin_metadata_files(root_dir=None, strict=False):
                 break
 
 
-def _get_plugin(name, enabled=None, builtin=None, check_for_duplicates=True):
-    plugin = None
-    for _plugin in _list_plugins(enabled=enabled, builtin=builtin):
-        if _plugin.name == name:
-            if check_for_duplicates and plugin is not None:
-                raise ValueError(f"Multiple plugins found with name '{name}'")
+def _get_plugin(name, enabled=None, builtin=None):
+    for plugin in _list_plugins(
+        enabled=enabled, builtin=builtin, shadowed=False
+    ):
+        if plugin.name == name:
```

Comment:
We use this style all over and its not that efficient. Instead of loading everything and then searching for a name we should load only until we get the named plugin. Mostly concerned about unnecessary fs calls.

Maybe a `_find_plugin(name=, plugin_dir=)` would be useful for this.

Can handle such improvements elsewhere.

Reply:
yes refactoring `_list_plugins()` to be a generator would 2x average performance here.

Importantly, however, I believe all of the App's plugin usage goes through `build_plugin_contexts()`, which is cached via `@plugins_cache`.